### ISSUE TITLE
Make precedence suggestion a "real" suggestion

### ISSUE THREAD

**Issue by killercup** [Manishearth/rust-clippy#1464]
Just saw this output:

```bash
$ cargo clippy
Compiling tokio-proto v0.1.0 (file:///Users/pascal/Projekte/tools/tokio-proto)
error: operator precedence can trip the unwary. Consider parenthesizing your expression:`self.allocated & (self.allocated - 1)`
   --> src/streaming/multiplex/frame_buf.rs:228:23
    |
228 |         debug_assert!(self.allocated & self.allocated - 1 == 0);
    |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
…
```

It would be lovely to have the "Consider parenthesizing your expression:`self.allocated & (self.allocated - 1)`" part be an actual suggestion (using `span_suggestion`) that can be applied by rustfix.

**Comment by llogiq** [comment]
When I wrote that lint, we had no `span_suggestion` yet. Good catch though, and quite easy to fix.

**Comment by killercup** [comment]
Damn, you are _fast_! I was gonna fix this myself this evening but you beat me to it! Keep up the great work!
